libraries/rush-lib/src/logic/pnpm/SubspacePnpmfileConfiguration.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { FileSystem, Import, type IPackageJson, JsonFile } from '@rushstack/node-core-library';
+import * as subspaceGlobalPnpmfile from './SubspaceGlobalPnpmfileShim';
+import { subspacePnpmfileShimFilename, scriptsFolderPath } from '../../utilities/PathConstants';
+
+import type { IPnpmfileContext, ISubspacePnpmfileShimSettings, IWorkspaceProjectInfo } from './IPnpmfile';
+import type { RushConfiguration } from '../../api/RushConfiguration';
+import type { PnpmPackageManager } from '../../api/packageManager/PnpmPackageManager';
+import { SubspacesConfiguration } from '../../api/SubspacesConfiguration';
+import { RushConstants } from '../RushConstants';
+
+/**
+ * Loads PNPM's pnpmfile.js configuration, and invokes it to preprocess package.json files,
+ * optionally utilizing a pnpmfile shim to inject preferred versions.
+ */
+export class SubspacePnpmfileConfiguration {
+  private _context: IPnpmfileContext | undefined;
+
+  public constructor(rushConfiguration: RushConfiguration) {
```

Comment:
@william2958  The `writeCommonTempSubspaceGlobalPnpmfileAsync()` static method is used, but as far as I can tell nothing ever constructs an instance of this class.

Is that intended? If so, then we should remove the constructor and  `transform()` methods.